packages/patternfly-3/patternfly-react/src/components/LoginPage/components/LoginCardComponents/LoginCardWithValidation.js

```diff
@@ -44,6 +44,11 @@ class LoginCardWithValidation extends React.Component {
         showError: false
       }
     });
+    this.setState(prevState => ({
+      form: {
+        disableSubmit: prevState.usernameField.value.length < 1 || prevState.passwordField.value.length < 1
+      }
+    }));
```

Comment:
I think it should be up to the application to decide if the submit button is disabled or not. We have the correct properties setup but we are not updating state when the properties are updated. We should have a getDerivedStateFromProps method that updates the state if the props have changed. 

That said, the class currently does not use that pattern and updating the props currently has no effect on the component. I guess this change is consistent with that and can be accepted. Changing the entire class to update only if the application updates the props would likely break any existing uses of the component.

Reply:
I agree we can add `getDerivedStateFromProps` mechanism here anyway,
@jeff-phillips-18 you think it will really break things for us? 

most of the updates have been called by callbacks until now from components that are wrapped by this container,
and your suggestion adds the ability to update it from its parent, right?

We are still waiting on `patternfly-design` decision over this design change :)

Reply:
If we update to rely on the parent to set the props where we were not before then it would be a breaking change since existing uses wouldn't update the props.

Reply:
I've added `getDerivedStateFromProps`, it will change the state for non validated login card.
For validated login card the disable submit state will only update on input change.